ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -336,18 +337,17 @@ public StreamPullQueryMetadata createStreamPullQuery(
     }
 
     // Stream pull query overrides.
-    final ConfiguredStatement<Query> statement = statementOrig.withConfigOverrides(
-        ImmutableMap.<String, Object>builder()
-            .putAll(statementOrig.getSessionConfig().getOverrides())
-            // Starting from earliest is semantically necessary.
-            .put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
-            // Using a single thread keeps these queries as lightweight as possible, since we are
-            // not counting them against the transient query limit.
-            .put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
-            // There's no point in EOS, since this query only produces side effects.
-            .put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.AT_LEAST_ONCE)
-            .build()
-    );
+    final Map<String, Object> overrides =
+        new HashMap<>(statementOrig.getSessionConfig().getOverrides());
+    // Starting from earliest is semantically necessary.
+    overrides.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    // Using a single thread keeps these queries as lightweight as possible, since we are
+    // not counting them against the transient query limit.
+    overrides.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1);
+    // There's no point in EOS, since this query only produces side effects.
+    overrides.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.AT_LEAST_ONCE);
+
+    final ConfiguredStatement<Query> statement = statementOrig.withConfigOverrides(overrides);
```

Comment:
The bug was that I assumed the ImmutableMap builder would let me put in the desired overrides after the `putAll` (like a regular map) and then become immutable only after you call `build`, but it turns out it actually checks that you put each key only once.

I just went ahead and made a copy of the input map (so it will be mutable) and then did the overrides.